L24-c++-extensible-api.md

```diff
@@ -0,0 +1,61 @@
+L24: C++ Extensible api
+-----------------------
+* Author(s): makdharma
+* Approver: vjpai
+* Status: Draft
+* Implemented in: https://github.com/grpc/grpc/pull/14517
+* Last updated: Feb 26, 2018
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+This proposal will enable extending functionality of GRPC C++ API by removing
+the ```final``` keyword and making some class methods ```virtual```. It will allow
+alternate implementations of the API by subclassing the API classes.
+
+## Background
+
+All GRPC C++ API classes are currently marked ```final```. The rationale is that
+having virtual methods might have adverse memory size and performance impact.
+Additionally, adding ```final``` to public API classes later on would break existing
+implementations that use classes derived from public API. Hence it is better and
+safer to mark classes ```final``` unless there's a reason to make them extensible.
+
+### Related Proposals:
+
+## Proposal
+
+* Remove ```final``` keyword from all public API classes.
+* Move current private core methods and member variables to protected.
```

Comment:
Member variables cannot be `protected` (or `public`) according to the C++ style guide. Consider only moving the methods or adding setters/getters or something like that.

Reply:
Also, what do you mean by "core" here and below?

Reply:
Core is a misnomer, I changed the text. A subset of methods is what I wanted to say.